spec/unit/inventory_spec.rb

```diff
@@ -52,6 +66,31 @@
     end
   end
 
+  describe '#metrics' do
+    it 'returns top level metrics' do
+      expect(resource.metrics.size).to eq(3)
+    end
+
+    it 'returns all metrics on the tree' do
+      expect(resource.metrics(true).size).to eq(11)
+    end
+
+    it 'top level metrics are of type Metric' do
+      expect(resource.metrics).to all(be_a(::Hawkular::Inventory::Metric))
+    end
+
+    it 'all metrics on the tree are of type Metric' do
+      expect(resource.metrics(true)).to all(be_a(::Hawkular::Inventory::Metric))
+    end
+
+    describe '#metrics_by_type' do
```

Comment:
Is it intentional to put it inside `describe '#metrics'` block, rather than on the same level?

Reply:
No. My bad.